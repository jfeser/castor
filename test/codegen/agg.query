select([(1.0 + 2.0) as x, avg(a) as y, count() as z, sum(a) as r, min(a) as s, max(a) as t],
  alist(r2, ascalar(0.a)))