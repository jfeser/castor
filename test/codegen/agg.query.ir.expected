// Locals:
// hoisted0 : fixed (persists=false)
// hoisted1 : fixed (persists=false)
// i30 : int (persists=true)
// cstart29 : int (persists=true)
// max28 : fixed (persists=false)
// min27 : fixed (persists=false)
// sum26 : fixed (persists=false)
// tup21 : tuple[fixed] (persists=false)
// found_tup22 : bool (persists=false)
// avg_num24 : fixed (persists=false)
// count25 : int (persists=false)
// avg_dem23 : int (persists=false)
// count31 : int (persists=true)
fun printer () : void {
    hoisted0 = int2fl(100);
    hoisted1 = 1.0 + 2.0;
    found_tup22 = false;
    avg_dem23 = 0;
    avg_num24 = 0.0;
    count25 = 0;
    sum26 = 0.0;
    min27 = 4611686018427387903.0;
    max28 = -4611686018427387904.0;
    cstart29 = 0;
    i30 = 0;
    count31 = 5;
    loop (i30 < count31) {
        tup21 = (int2fl(buf[cstart29 : 2]) / hoisted0);
        avg_num24 = avg_num24 + tup21[0];
        avg_dem23 = avg_dem23 + 1;
        count25 = count25 + 1;
        sum26 = sum26 + tup21[0];
        min27 = tup21[0] < min27 ? tup21[0] : min27;
        max28 = tup21[0] < max28 ? max28 : tup21[0];
        found_tup22 = true;
        cstart29 = cstart29 + 2;
        i30 = i30 + 1;
    }
    if (found_tup22) {
        print(tuple[fixed, fixed, int, fixed, fixed, fixed],
        (hoisted1, avg_num24 / int2fl(avg_dem23), count25, sum26, min27,
         max28));
    } else {
         
    }
}
// Locals:
// hoisted2 : fixed (persists=false)
// hoisted3 : fixed (persists=false)
// avg_dem7 : int (persists=false)
// count9 : int (persists=false)
// i14 : int (persists=true)
// max12 : fixed (persists=false)
// count15 : int (persists=true)
// avg_num8 : fixed (persists=false)
// sum10 : fixed (persists=false)
// cstart13 : int (persists=true)
// tup5 : tuple[fixed] (persists=false)
// found_tup6 : bool (persists=false)
// min11 : fixed (persists=false)
fun consumer () : void {
    hoisted2 = int2fl(100);
    hoisted3 = 1.0 + 2.0;
    found_tup6 = false;
    avg_dem7 = 0;
    avg_num8 = 0.0;
    count9 = 0;
    sum10 = 0.0;
    min11 = 4611686018427387903.0;
    max12 = -4611686018427387904.0;
    cstart13 = 0;
    i14 = 0;
    count15 = 5;
    loop (i14 < count15) {
        tup5 = (int2fl(buf[cstart13 : 2]) / hoisted2);
        avg_num8 = avg_num8 + tup5[0];
        avg_dem7 = avg_dem7 + 1;
        count9 = count9 + 1;
        sum10 = sum10 + tup5[0];
        min11 = tup5[0] < min11 ? tup5[0] : min11;
        max12 = tup5[0] < max12 ? max12 : tup5[0];
        found_tup6 = true;
        cstart13 = cstart13 + 2;
        i14 = i14 + 1;
    }
    if (found_tup6) {
        consume(tuple[fixed, fixed, int, fixed, fixed, fixed],
        (hoisted3, avg_num8 / int2fl(avg_dem7), count9, sum10, min11, max12));
    } else {
         
    }
}
